### ISSUE TITLE
Unwrapping / extending inline structs 

### ISSUE THREAD

**Issue by 0x-buidl** [gzuidhof/tygo#28]
Structs that are tagged as `inline` should be unwrapped. 

```go
// Input
type Vehicle struct {
	Category string `json:"category"`
	Make     string `json:"make"`
}

type Toyota struct {
	Vehicle `    json:",inline" tstype:",inline"`
	Year    int `json:"year"`
}
```

```ts
//Output
interface Vehicle {
  category: string;
  make: string;
}

interface Toyota extends Vehicle {
  year: number;
}

//Or 
interface Toyota  {
  category: string;
  make: string;
  year: number;
}
```

**Comment by gzuidhof** [comment]
I think this is the same underlying challenge as in #15. It can definitely be done, but the current tygo transpiler is single pass, which makes this tricky (if not impossible).

**Comment by 0x-buidl** [comment]
@gzuidhof  made a PR for this #30 .

**Comment by gzuidhof** [comment]
Shipped with #30, thank you @0x-buidl
